CMakeLists.txt

```
@@ -29,6 +29,14 @@ elseif (“${CMAKE_CXX_COMPILER_ID}” STREQUAL “Intel”)
     set(CMAKE_CXX_FLAGS_RELEASE "-O2 -xHost -m64 -pipe")
 endif()
 
+include(CheckIPOSupported)
+check_ipo_supported(RESULT supported OUTPUT error)
```

Maybe you could also specify C++?
```suggestion
check_ipo_supported(RESULT supported OUTPUT error LANGUAGES CXX)
```